src/app/beer_garden/requests.py

```diff
@@ -735,3 +736,8 @@ def handle_event(event):
                 setattr(existing_request, field, getattr(event.payload, field))
 
             db.update(existing_request)
+
+    # Required if the main thread spawns a wait Request
+    if event.name == Events.REQUEST_COMPLETED.name:
+        if str(event.payload.id) in request_map:
+            request_map[str(event.payload.id)].set()
```

Comment:
OK, I think I understand why we have to do this here. We also do this in the `complete_request` method, but that was based on the assumption that the same entry point (aka process) would be used to create the request and to complete it.

That assumption doesn't hold for this case, because the main process is creating the request so the request_map entry lives there. Then the HTTP entry point is handling the completion and is unable to find it.

I'm going to see if removing the old code works now that you've added this so we don't have duplicate functionality laying around.

Also, we should probably make this shared memory or something so we don't have to think about this anymore :thinking:

Reply:
Should have thought about it more. Until it's shared memory we actually need it in both.

Reply:
We can make that a future task for shared memory, until then, I will keep this in here.